src/core/main.cpp

```diff
@@ -841,6 +842,86 @@ void print_env_vars_xlio_global_settings()
                 "---------------------------------------------------------------------------\n");
 }
 
+/**
+  * @brief Shows configured values using log messages
+  * @param key Configuration parameter name
+  * @param param_descriptor Configuration parameter descriptor. nullptr for sub-objects.
+  * @param element Current value for the parameter, as any
+  * @param def_value Default value for the parameter, as any. When there is no defauilt value defined, it is empty
+  */
+void print_config_element(const std::string& key, const parameter_descriptor* param_descriptor, 
+                           std::experimental::any element, std::experimental::any def_value_any)
+{
+    const std::string title = param_descriptor ? param_descriptor->get_title() : "";
+    const std::type_info& type = element.type();
+    
+    if (type == typeid(bool)) {
+        bool def_value = (def_value_any.type() == typeid(bool)) ? std::experimental::any_cast<bool>(def_value_any) : false;
+        bool cur_value = std::experimental::any_cast<bool>(element);
+        VLOG_PARAM_BOOL(title.c_str(), cur_value, def_value, key.c_str());
+    }
+    else if (type == typeid(std::string)) {
+        std::string def_value = (def_value_any.type() == typeid(std::string)) ? std::experimental::any_cast<std::string>(def_value_any) : "";
+        std::string cur_value = std::experimental::any_cast<std::string>(element);
+        VLOG_PARAM_STRING(title.c_str(), cur_value, def_value, key.c_str(), cur_value.c_str());
+    }
+    else if (type == typeid(int64_t)) {
+        int64_t def_value = (def_value_any.type() == typeid(int64_t)) ? std::experimental::any_cast<int64_t>(def_value_any) : 0;
```

Comment:
i think it's worth splitting to print_int64_element, and for a clean symmetry, create print_bool_element, etc.
(imho function is too long)